src/loc.rs

```diff
@@ -64,12 +64,10 @@ impl LocDiff {
             Some(removed.parse().map_err(LocParseError::RemovedParseError)?)
         };
 
-        let file = file.into();
-
         Ok(Self {
             added,
             removed,
-            file,
+            file: String::from(file),
```

Comment:
Why are you doing this?

Reply:
It should have been shorter but I can revert it. :wrench:

Reply:
Hmm, I think not, that it is worth saving that one line. I think readability is preferable over shortness always.

Reply:
In general, I prefer functional programming whenever and wherever possible.  This is the more functional style which you also apply for the original definition of your `Commit` constructor.  Since I oriented on this definition when I started to refactor your code, I decided to rewrite this constructor as shown.

Reply:
And, by the way, I consider this way more readable than this one line `let file = file.into();` before the call of the constructor.

Reply:
> In general, I prefer functional programming whenever and wherever possible. This is the more functional style which you also apply for the original definition of your `Commit` constructor. Since I oriented on this definition when I started to refactor your code, I decided to rewrite this constructor as shown.

Fair enough, you got me there. I guess, I've not yet developed a style here..

> And, by the way, I consider this way more readable than this one line `let file = file.into();` before the call of the constructor.

I see, where you coming from, in this regard having an IDE is kinda expected from the user, I guess 😅